operational_analysis/methods/quality_check_automation.py

```diff
@@ -0,0 +1,331 @@
+        dst_df = pd.read_csv('./daylight_savings.csv')
+
+        self._df_dst =  self._df.loc[self._df[self._id]==self._df[self._id].unique()[0], :]
+    
+        df_full = timeseries.gap_fill_data_frame(self._df_dst, self._t, self._freq) # Gap fill so spring ahead is visible
+        df_full.set_index(self._t, inplace=True)
+    
+        years = df_full.index.year.unique() # Years in data record
+        num_years = len(years)
+    
+        plt.figure(figsize = (12,20))
+
+        for y in np.arange(num_years):
+            dst_data = dst_df.loc[dst_df['year'] == years[y]] 
+        
+            # Set spring ahead window to plot
+            spring_start = pd.to_datetime(dst_data['start']) - pd.Timedelta(hours = hour_window)
+            spring_end = pd.to_datetime(dst_data['start']) + pd.Timedelta(hours = hour_window)
+        
+            # Set fall back window to plot
+            fall_start = pd.to_datetime(dst_data['end']) - pd.Timedelta(hours = hour_window)
+            fall_end = pd.to_datetime(dst_data['end']) + pd.Timedelta(hours = hour_window)
+        
+            # Get data corresponding to each
+            data_spring = df_full.loc[(df_full.index > spring_start.values[0]) & (df_full.index < spring_end.values[0])]
+            data_fall = df_full.loc[(df_full.index > fall_start.values[0]) & (df_full.index < fall_end.values[0])]
+
+            # Plot each as side-by-side subplots
+            plt.subplot(num_years, 2, 2*y + 1)
+            if np.sum(~np.isnan(data_spring[self._w])) > 0:
+                plt.plot(data_spring[self._w])
+            plt.title(years[y])
+        
+            plt.subplot(num_years, 2, 2*y + 2)
+            if np.sum(~np.isnan(data_fall[self._w])) > 0:
+                plt.plot(data_fall[self._w])
+            plt.title(years[y])
+    
+        plt.tight_layout()
+        plt.show()
+
+    def max_min(self):
+
+        """
+        Creates a DataFrame that contains the max and min values for each column
+
+        Args:
+        (None)
+
+        Returns:
+        (None)
+        """
+
+        self._max_min = pd.DataFrame(index = self._df.columns, columns = {'max', 'min'})
+        self._max_min['max'] = self._df.max()
+        self._max_min['min'] = self._df.min()
+
+    def turb_plots(self):
+
+        """
+        Produces power curve (power vs. wind speed) plot for each individual turbine
+
+        Args:
+        (None)
+
+        Returns:
+        (None)
+        """
+        turbs = self._df[self._id].unique()
+        num_turbs = len(turbs)
+        num_rows = np.ceil(num_turbs/4.)
+
+        plt.figure(figsize = (15,num_rows * 5))
+        n = 1
+        for t in turbs:
+            plt.subplot(num_rows, 4, n)
+            scada_sub = self._df.loc[self._df[self._id] == t, :]
+            plt.scatter(scada_sub[self._ws], scada_sub[self._w], s = 5)
+            n = n + 1
+            plt.title(t)
+        plt.tight_layout()
+        plt.show()
+
+    def column_plots(self):
```

Comment:
Not sure what column plot means. We're plotting histograms for every data field in a df, right?

Reply:
Yes; every numeric column (i.e. int or float). I can change the name to be more intuitive.